src/renderers/shared/fiber/__tests__/ReactIncrementalErrorHandling-test.js

```diff
@@ -275,4 +275,24 @@ describe('ReactIncrementalErrorHandling', () => {
     expect(ReactNoop.getChildren('e')).toEqual(null);
     expect(ReactNoop.getChildren('f')).toEqual(null);
   });
+
+  it('restores environment state despite error during commit', () => {
```

Comment:
Ideally we'd have a DOM specific test for this that tests the observable intent.